internal/dataplane/parser/translate_httproute_test.go

```diff
@@ -1479,3 +1371,18 @@ func TestIngressRulesFromHTTPRoutes_RegexPrefix(t *testing.T) {
 func HTTPMethodPointer(method string) *gatewayv1beta1.HTTPMethod {
 	return (*gatewayv1beta1.HTTPMethod)(&method)
 }
+
+func k8sObjectInfoOfHTTPRoute(route *gatewayv1beta1.HTTPRoute) util.K8sObjectInfo {
+	// parsers always provide the annotations map
+	anotations := make(map[string]string)
+	if route.Annotations != nil {
+		anotations = route.Annotations
+	}
+
+	return util.K8sObjectInfo{
+		Name:             route.Name,
+		Namespace:        route.Namespace,
+		Annotations:      anotations,
+		GroupVersionKind: route.GroupVersionKind(),
```

Comment:
Yes it does. Please take a not that tests are passing. If it did not, the tests would fail, as it's doing an equality comparision.

Reply:
And it does so, because the test logic itself sets `httproute.SetGroupVersionKind(httprouteGVK)`. The expect function is executed afterwards.

Reply:
Alright so this is nice but what if someone was to write a new test and he/she wasn't aware that this has to be filled in 🤔 ? If the test doesn't explicitly check nor set the GVK this they might expect a different behaviour, right?

Reply:
Well, the parses uses `objectInfo := util.FromK8sObject(httproute)`, so i may as well use that here too.

If one does not set GVK in his tests, it's more that's going to change, i.e the route name.

The intent of the code is to set the `kongstate.Ingress` based on route. And the tests does that. But it does that in a crude way, defining the same  kongstate.Ingress for bunch of tests.

The other possibility I see here, is to provide asserts/checks in the test function instead of providing desired output in form of `ingressRules` internal object. Providing whole `ingressRules` is annoying.

Reply:
My suggestion to move forward with this: since this function accepts only an http route, set a gvk just for that in the same way it was set before this change. This way we retain the original behavior and introduce a helper to reduce the boilerplate (as I believe was the original intent)

Reply:
Sure, can do.

Reply:
@pmalek ptal